api/envoy/config/core/v3/address.proto

```diff
@@ -128,6 +128,10 @@ message BindConfig {
   // Additional socket options that may not be present in Envoy source code or
   // precompiled binaries.
   repeated SocketOption socket_options = 3;
+
+  // Additional source addresses can be used to bind when creating a socket. The
+  // source address selection is based on the target host address' version.
+  repeated SocketAddress additional_source_addresses = 4;
```

Comment:
From a high-level, it's not clear how this new field interacts with the `source_address` field above. Does it override it or adds more choices for source addresses?

Reply:
It adds more choices. I can clarify this in the comments